### ISSUE TITLE
feature_request(config): automatically detect configuration files

### ISSUE THREAD

**Issue by Kristinita** [igorshubovych/markdownlint-cli#50]
### 1. Summary

It would be nice, if markdownlint-cli will automatically detect configuration files as [**`.markdownlint.yaml`**](https://github.com/DavidAnson/markdownlint/issues/118#issuecomment-382098978).

### 2. Argumentation

Save a users time. It would be nice, if user doesn't need add `-c .markdownlint.yaml` each time.

I use [**many linters**](https://kristinita.ru/Pages/About-for-developers#validation-table) in my projects: all except Markdownlint detect standard config files.

### 3. Data

+ `KiraMarkdownlint__MD002_MD041--demo.md`:

```markdown
### Kira

Goddess!

```

+ `.markdownlint.yaml`:

```yaml
MD041: false
```

### 4. Steps to reproduce

```text
markdownlint KiraMarkdownlint__MD002_MD041--demo.md
```

### 5. Expected behavior

As if the configuration file were directly specified:

```text
D:\Kristinita>markdownlint KiraMarkdownlint__MD002_MD041--demo.md -c .markdownlint.yaml
KiraMarkdownlint__MD002_MD041--demo.md: 1: MD002/first-heading-h1/first-header-h1 First heading should be a top level heading [Expected: h1; Actual: h3]
```

### 6. Actual behavior

```text
D:\SashaDebugging>markdownlint KiraMarkdownlint__MD002_MD041--demo.md
KiraMarkdownlint__MD002_MD041--demo.md: 1: MD002/first-heading-h1/first-header-h1 First heading should be a top level heading [Expected: h1; Actual: h3]
KiraMarkdownlint__MD002_MD041--demo.md: 1: MD041/first-line-h1 First line in file should be a top level heading [Context: "### Kira"]
```

### 7. Environment

+ Windows 10 Enterprise LTSB 64-bit EN
+ Node.js 11.9.0
+ markdownlint-cli 0.13.0

Thanks.

**Comment by igorshubovych** [comment]
Thanks for suggestion @Kristinita

**Comment by igorshubovych** [comment]
@Kristinita actually if it is `.markdownlint.json` it should just work.

Internally it uses `rc` (https://github.com/dominictarr/rc).

**Comment by DavidAnson** [comment]
FYI, I added support for automatic YAML (and now YML) config files to the VS Code extension, so this is a small feature gap. I’m due to update the CLI soon and will have a look if you don’t get to this.

**Comment by Kristinita** [comment]
**Status: ✔️ Fixed for me**

Thanks!
